main.py

```diff
@@ -4,8 +1,52 @@
+import json
+from db.models import Race, Skill, Guild, Player
+from typing import Any, Optional
 
 
 def main() -> None:
-    pass
+    with open("players.json", "r") as file:
+        players_data: dict[str, Any] = json.load(file)
+
+    for player_name, player_data in players_data.items():
+        if not isinstance(player_data, dict):
+            continue
+
+        race_data: Optional[dict[str, Any]] = player_data.get("race")
```

Comment:
The use of type hints like 'Optional[dict[str, Any]]' is valid in Python 3.9+, but ensure your environment supports this syntax. If you're targeting earlier versions, consider using 'Optional[Dict[str, Any]]' from the 'typing' module.